homeassistant/components/enphase_envoy/__init__.py

```diff
@@ -1 +1,100 @@
-"""The enphase_envoy component."""
+"""The Enphase Envoy integration."""
+from __future__ import annotations
+
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from envoy_reader.envoy_reader import EnvoyReader
+import httpx
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import COORDINATOR, DOMAIN, NAME, PLATFORMS, SENSORS
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Enphase Envoy from a config entry."""
+
+    config = entry.data
+    name = config[CONF_NAME]
+    envoy_reader = EnvoyReader(
+        config[CONF_HOST], config[CONF_USERNAME], config[CONF_PASSWORD]
+    )
+
+    try:
+        await envoy_reader.getData()
+    except httpx.HTTPStatusError as err:
+        _LOGGER.error("Authentication failure during setup: %s", err)
+        return
+    except (AttributeError, httpx.HTTPError) as err:
+        raise ConfigEntryNotReady from err
+
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        data = {}
+        async with async_timeout.timeout(30):
+            try:
+                await envoy_reader.getData()
+            except httpx.HTTPError as err:
+                raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+            for condition in SENSORS:
+                if condition != "inverters":
+                    data[condition] = await getattr(envoy_reader, condition)()
+                else:
+                    data["inverters_production"] = await getattr(
+                        envoy_reader, "inverters_production"
+                    )()
```

Comment:
```suggestion
                    data["inverters_production"] = await envoy_reader.inverters_production()
```

Reply:
Wow didn't notice that when I moved it.  That's weird